pkg/cluster/arooperator.go

```diff
@@ -5,27 +5,39 @@ package cluster
 
 import (
 	"context"
-
-	"github.com/Azure/ARO-RP/pkg/operator/deploy"
 )
 
+func (m *manager) isDefaultIngressProfileAvailable() bool {
+
+	for _, ip := range m.doc.OpenShiftCluster.Properties.IngressProfiles {
+		if ip.Name == "default" {
+			return true
+		}
+	}
```

Comment:
Hm. I just realised that we should probably skip only if `len(m.doc.OpenShiftCluster.Properties.IngressProfiles) == 0`. We don't use the default profile (so cheking default profile is confusing). We only use the first ingress profile [here](https://github.com/Azure/ARO-RP/blob/907753fbd1b32ff78b234997caca06ee6bd9b842/pkg/operator/deploy/deploy.go#L171).

But if we are going to re-use this function as suggested in https://github.com/Azure/ARO-RP/pull/1972/files#r809928705 - we need to make sure that it is also the case for `createOrUpdateRouterIPFromCluster`.

Reply:
as discussed , the log was adapted to skip when len(m.doc.OpenShiftCluster.Properties.IngressProfiles) == 0, that includes the nil case. Comment added to explain the background